File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector <int> ans(nums.size());
        long prod = 1;
        for (int i = 0; i < nums.size(); ++i){
            ans[i] = prod;
            prod *= nums[i];
        }
        prod = 1;
        for (int i = nums.size() - 1; i > -1; --i){
            ans[i] *= prod;
            prod *= nums[i];
        }
        return ans;
    }
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prod = 1
        N = len(nums)
        ans = [0]*N
        for i, n in enumerate(nums):
            ans[i] = prod
            prod *= n
        prod = 1
        for i in range(N - 1, -1, -1):
            ans[i] *= prod
            prod *= nums[i]
        return ans